### PR TITLE
fix compiler warning and add option to control systemd service file installation

### PR DESCRIPTION
**Pull request by abhimanyuv1**

### CODE REVIEW

File: core/src/common/lwm2m_tlv.c [hunk 1]
```diff
@@ -341,12 +341,16 @@ static int TlvEncodeFloat(uint8_t * buffer, int bufferLen, int type, int id, dou
     if (floatSize == 4)
     {
         float f = value;
-        int32_t temp = htonl(*(uint32_t*)&f);
+        int32_t temp;
```
Comment (boyvinall) [inline core/src/common/lwm2m_tlv.c]: Were you having alignment problems using the pointer addressing before?

Comment (boyvinall) [inline core/src/common/lwm2m_tlv.c]: Also, this is kind of separate to the systemd thing, but fine.

**Comment by boyvinall** [comment]
retest AWS CI (erbium)

**Comment by boyvinall** [comment]
retest AWS CI (libcoap)

**Comment by boyvinall** [comment]
@abhimanyuv1 sorry, could you re-merge please? thanks

**Comment by abhimanyuv1** [comment]
hi  matt, I had some comment in mail but cant find them here, if you want i can send 2 pull request for those patches?
also, i got these warning when i was creating buildroot packages and -Wall was enabled.

**Comment by boyvinall** [comment]
retest AWS CI (erbium)

**Comment by boyvinall** [comment]
retest AWS CI (libcoap)

**Pull request merged**